rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/dap/SuspendedState.java

```diff
@@ -0,0 +1,108 @@
+package org.rascalmpl.vscode.lsp.dap;
+
+import io.usethesource.vallang.ISourceLocation;
+import org.rascalmpl.debug.IRascalFrame;
+import org.rascalmpl.interpreter.Evaluator;
+import org.rascalmpl.interpreter.result.IRascalResult;
+import org.rascalmpl.uri.URIUtil;
+import org.rascalmpl.vscode.lsp.dap.variable.RascalVariable;
+import org.rascalmpl.vscode.lsp.dap.variable.VariableSubElementsCounter;
+import org.rascalmpl.vscode.lsp.dap.variable.VariableSubElementsCounterVisitor;
+import org.rascalmpl.vscode.lsp.dap.variable.VariableSubfieldsVisitor;
+
+import java.util.*;
+
+/**
+ * Class used to store the state of the Rascal Evaluator when it is suspended
+ */
+public class SuspendedState {
+
+    private final Evaluator evaluator;
+    private IRascalFrame[] currentStackFrames;
+    private final HashMap<Integer, RascalVariable> variables;
+    private final HashMap<Integer, IRascalFrame> scopes;
+    private int referenceIDCounter;
```

Comment:
make these non-final fields `atomic` as we don't want caching between threads. and lsp4j will be called from different threads.